### PR TITLE
Use Composer autoload to automatically include the needed classes

### PR DESCRIPTION
**Pull request by marco-c**
@staabm

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -23,5 +23,8 @@
         "mozilla/wp-sw-manager": "dev-master",
         "marco-c/wp_serve_file": "dev-master"
     },
+    "autoload": {
+      "classmap": ["vendor/"]
```
Comment (staabm) [inline composer.json]: no thats not correct... here you should define where "your" classes are placed, not those of your dependencies.

e.g. `wp-web-push/`

Comment (marco-c) [inline composer.json]: If I do this and https://github.com/marco-c/wp-web-app-manifest-generator/pull/9 and run `composer dumpautoload`, the generated `autoload_classmap.php` is empty.

Comment (marco-c) [inline composer.json]: The classes of the dependencies should be defined if they don't support the PSR-4 standard.

**Comment by marco-c** [comment]
The only confusing thing is that this doesn't work if I generate the "normal" (non-optimized) autoload code.

**Comment by staabm** [comment]
The reason is that you classnames/filenames are neither psr0, 2 nor 4 compilant

File: composer.json [hunk 2]
```diff
@@ -23,5 +23,8 @@
         "mozilla/wp-sw-manager": "dev-master",
         "marco-c/wp_serve_file": "dev-master"
     },
-    "minimum-stability": "dev"
+    "minimum-stability": "dev",
+    "autoload": {
+        "classmap": [ "vendor/mozilla/wp-sw-manager/" ]
```
Comment (staabm) [inline composer.json]: I dont think this is correct

**Comment by marco-c** [comment]
> The reason is that you classname are neither psr0, 2 nor 4 compilant

I see, I didn't think there was a standard for class names as well. I wanted to use the WordPress convention (WP_Class_Name), but I don't mind switching to the psr-4 standard.

**Comment by marco-c** [comment]
Should the file of the class have the same name as the class?

**Comment by staabm** [comment]
See http://www.php-fig.org/psr/psr-0/ or http://www.php-fig.org/psr/psr-4/ if you like less deep folder structures

File: composer.json (hunk 2 above)
Comment (marco-c) [inline composer.json]: https://getcomposer.org/doc/04-schema.md#classmap

This has to be done when the class doesn't support PSR-4.

Comment (staabm) [inline composer.json]: In case you decide for one of e psr standards you need to name it here instead of "classmap"

Comment (marco-c) [inline composer.json]: Yes, I will switch that class to the PSR-4 standard, for now I'm updating the other two classes we're using.

Thank you for the help by the way!

**Comment by marco-c** [comment]
e1e5fc3 updated the WP_Serve_File dependency (now it actually uses PSR-4).
Now, everything works also if I remove the `optimize-autoloader` option. I will keep it anyway since it's an optimization.

**Comment by marco-c** [comment]
https://github.com/mozilla/wp-add-to-homescreen/pull/53 is the PR for wp-add-to-homescreen, https://github.com/mozilla/wp-offline-shell/pull/100 for wp-offline-shell.

**Pull request merged**